crates/storage/provider/src/db_provider/storage.rs

```diff
@@ -46,11 +41,18 @@ impl<DB: Database> StateProviderFactory for ProviderImpl<DB> {
         let block_number =
             tx.get::<tables::HeaderNumbers>(block_hash)?.ok_or(Error::BlockHash { block_hash })?;
 
+        // check if block is canonical or not. Only canonical blocks have changesets.
+        let canonical_block_hash = tx
```

Comment:
Was this a bug / missing check? Let's pull in separate PR and a test for this case if so?

Reply:
This is result of removing hash from  `BlockTransition` (Index was `BlockNumHash`) table. It is better to be sure if given hash is in canonical block or not